libs/core/src/lib/button/button.component.ts

```diff
@@ -47,34 +55,45 @@ export class ButtonComponent extends AbstractFdNgxClass {
 
     /** @hidden */
     _setProperties() {
-        if (this.fdButtonClass) {
-            this._addClassToElement('fd-button');
-        }
-        if (this.compact) {
-            this._addClassToElement('fd-button--compact');
-        }
-        if (this.glyph) {
-            this._addClassToElement('sap-icon--' + this.glyph);
-        }
-        if (this.fdType) {
-            this._addClassToElement('fd-button--' + this.fdType);
-        }
-        if (this.options) {
-            if (typeof this.options === 'string') {
-                this._addClassToElement('fd-button--' + this.options);
-            } else if (Array.isArray(this.options)) {
-                this.options.forEach(option => {
-                    if (typeof option === 'string') {
-                        this._addClassToElement('fd-button--' + option);
-                    }
-                });
-            }
-        }
+
     }
 
     /** @hidden */
-    constructor(private elementRef: ElementRef) {
-        super(elementRef);
+    constructor(private _elementRef: ElementRef) {
+    }
+
+    ngOnInit() {
```

Comment:
Let's add return type and some comments to functions on this file.

Reply:
done